### ISSUE TITLE
Do you find that this action works to whitelist runner ips?

### ISSUE THREAD

**Issue by iDVB** [haythem/public-ip#10]
Right now, I'm getting IPs like `52.165.42.83`. I then whitelist in another step, and then have a lighthouse test in the next step. 
Lighthouse then reports `403` which is what happens when the WAF denies rather than allows. Then googling around there seems to be different ways on how to properly whitelist GitHub Actions:

* [public IP test](https://github.com/haythem/public-ip)
* [getting from github meta API call](https://api.github.com/meta) under `actions`. This is actually in [the docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#ip-addresses).
* Not possible without using **self-hosted runners**. This is also in those same [docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#ip-addresses).
* Pulling from Azure's public docs that get updated. Mentioned by a [partner in the forums](https://github.community/t/stable-ip-addresses-range-for-actions/16515/2).

Needless to say, I'm a bit confused as to what info is the latest and what actually works **now**.

**Comment by tlhunter** [comment]
I found this did work to whitelist runner IPs. However, it seems to have stopped working a few hours ago...

**Comment by haythem** [comment]
Hello @iDVB,

I'm sorry for the late response. I think the action works like expected. It returns the runner's public IP address. All you have to do is to use the tool that works with your target environment (Azure, AWS, etc.) and authorize the returned address by the action.

A reproduction repo / workflow is needed so I can help you.

**Comment by astrojuanlu** [comment]
Indeed, from https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#ip-addresses

> Since there are so many IP address ranges for GitHub-hosted runners, we do not recommend that you use these as allowlists for your internal resources. Instead, we recommend you use larger runners with a static IP address range, or self-hosted runners. For more information, see [Using larger runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners) or [About self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners).
> The list of GitHub Actions IP addresses returned by the API is updated once a week.
